scorecards.go

```diff
@@ -0,0 +1,130 @@
+package opslevel
+
+import (
+	"fmt"
+)
+
+type ScorecardId struct {
+	Aliases []string `graphql:"aliases"`
+	Id      ID       `graphql:"id"`
+}
+
+type Scorecard struct {
+	ScorecardId
+
+	Description   string      `graphql:"description"`
+	Filter        Filter      `graphql:"filter"`
+	Name          string      `graphql:"name"`
+	Owner         EntityOwner `graphql:"owner"`
+	PassingChecks int         `graphql:"passingChecks"`
+	ServiceCount  int         `graphql:"serviceCount"`
+	TotalChecks   int         `graphql:"totalChecks"`
+}
+
+type ScorecardConnection struct {
+	Nodes      []Scorecard `graphql:"nodes"`
+	PageInfo   PageInfo    `graphql:"pageInfo"`
+	TotalCount int         `graphql:"totalCount"`
+}
+
+type ScorecardCheckConnection struct {
+	Nodes      []Check  `graphql:"nodes"`
+	PageInfo   PageInfo `graphql:"pageInfo"`
+	TotalCount int      `graphql:"totalCount"`
+}
+
+type ScorecardInput struct {
+	Name        string  `graphql:"name" json:"name"`
+	Description *string `graphql:"description" json:"description,omitempty"`
+	// TODO: API shouldn't use IdentifierInput for Owner if we support groups and teams
+	// if an alias is passed, is that for a Group or a Team?
+	// on Domain and System objects, they only support ID type as the owner, not InputIdentifier
+	// if this is changed, update the GetScorecard test
+	// TODO: a way to get linked checks from a scorecard
+	// requires a change in API and structs
+	Owner    IdentifierInput `graphql:"owner" json:"owner"`
```

Comment:
```suggestion
	Owner    ID `graphql:"owner" json:"ownerId"`
```

Reply:
Looks like we can't `omitempty` here, since the field is required in `ScorecardInput`

Reply:
Ok - then it should be a value type not a pointer type

Reply:
Good point, just changed it so that `ownerId` is no longer type `*ID`, it's now `ID`.